js/tuxexchange.js

```diff
@@ -0,0 +1,533 @@
+                'cancelOrder': true,
+                'fetchOpenOrders': true,
+                'fetchMyTrades': true,
+                'fetchDepositAddress': true,
+                'fetchWithdrawals': true,
+                'fetchDeposits': true,
+                'fetchClosedOrders': false,
+                'fetchL2OrderBook': false,
+                'fetchOHLCV': false,
+                'fetchOrder': false,
+                'editOrder': false,
+                'fetchTransactions': false,
+                'fetchLedger': false,
+                'withdraw': false,
+            },
+            'urls': {
+                'logo': 'https://www.tuxexchange.com/images/kittytrade.png',
+                'api': 'https://www.tuxexchange.com/api',
+                'www': 'https://www.tuxexchange.com',
+                'doc': 'https://www.tuxexchange.com/docs',
+                'fees': 'https://www.tuxexchange.com/faq',
+            },
+            'api': {
+                // All methods are passed in as query params
+                'public': { 'get': [''] },
+                'private': { 'post': [''] },
+            },
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': 0,
+                    'taker': 0.3,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const tickersData = await this.publicGet ({ 'method': 'getticker' });
+        // Would prefer not to make two API calls, but this is the only way to get fees for the market
+        const coinResponse = await this.publicGet ({ 'method': 'getcoins' });
+        const tickerIds = Object.keys (tickersData);
+        let result = [];
+        for (let i = 0; i < tickerIds.length; i++) {
+            const id = tickerIds[i];
+            const tickerData = tickersData[id];
+            const splitId = id.split ('_');
+            const baseId = splitId[0]; // base
+            const quoteId = splitId[1]; // quote
+            const base = this.commonCurrencyCode (baseId);
+            const quote = this.commonCurrencyCode (quoteId);
+            const active = this.safeInteger (tickerData, 'isFrozen') === 0;
+            const coinData = coinResponse[quoteId];
+            const market = {
+                'id': id,
+                'symbol': base + '/' + quote,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': active,
+                'maker': this.safeString (coinData, 'makerfee'),
+                'taker': this.safeString (coinData, 'takerfee'),
+                'info': tickerData,
+            };
+            result.push (market);
+        }
+        return result;
+    }
+
+    async fetchCurrencies (params = {}) {
+        const currenciesData = await this.publicGet ({ 'method': 'getcoins' });
```

Comment:
We need to propagate `params` to the exchange endpoint – everywhere throughout this file, as we do, for example, here: https://github.com/ccxt/ccxt/blob/master/js/binance.js#L848